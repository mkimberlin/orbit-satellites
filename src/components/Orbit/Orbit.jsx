import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './styles.module.css';

const displayName = 'Orbit';
const propTypes = {
  position: PropTypes.number
};

const Orbit = ({ children, position = 1 }) => (
  <div className={cx('orbit', `orbit-${position}`)}>{children}</div>
);

Orbit.displayName = displayName;
Orbit.propTypes = propTypes;

export default Orbit;
