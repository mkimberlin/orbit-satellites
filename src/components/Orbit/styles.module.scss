@import '../../styles/common-variables.scss';
@import '../../styles/animations.scss';
@import '../../styles/common-variables.scss';

@include keyframe(orbit) {
  0% {
    transform: rotateZ(0deg);
  }
  100% {
    transform: rotateZ(360deg);
  }
}

.orbit {
  position: absolute;
  top: 50%;
  left: 50%;
  border: 2px solid black;
  border-radius: 50%;
  transform-style: preserve-3d;

  @for $orbit from 1 through $max-orbits {
    &.orbit-#{$orbit} {
      width: calc(#{$orbit-base} + (#{$orbit-step} * #{$orbit}));
      height: calc(#{$orbit-base} + (#{$orbit-step} * #{$orbit}));
      margin-top: calc(
        -1 * (#{$orbit-base} + (#{$orbit-step} * #{$orbit})) / 2
      );
      margin-left: calc(
        -1 * (#{$orbit-base} + (#{$orbit-step} * #{$orbit})) / 2
      );
      @include animation(
        calc(#{$speed-base} + (#{$speed-step} * #{$orbit - 1})),
        orbit
      );
    }
  }
}
