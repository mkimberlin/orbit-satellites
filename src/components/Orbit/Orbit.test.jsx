import React from 'react';
import Orbit from './Orbit';

describe('Orbit component', () => {
  it('renders correctly with default props', () => {
    snapshot(<Orbit />);
  });

  it('renders correctly with alternate props', () => {
    snapshot(
      <Orbit position={2} onClick={jest.fn()}>
        <div>Stuff</div>
      </Orbit>
    );
  });
});
