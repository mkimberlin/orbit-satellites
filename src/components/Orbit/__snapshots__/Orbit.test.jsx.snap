// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orbit component renders correctly with alternate props 1`] = `
<div
  className="orbit orbit-2"
/>
`;

exports[`Orbit component renders correctly with default props 1`] = `
<div
  className="orbit orbit-1"
/>
`;
